core/store/src/trie/prefetching_trie_storage.rs

```diff
@@ -249,6 +285,17 @@ impl TriePrefetchingStorage {
 }
 
 impl PrefetchStagingArea {
+    fn new(shard_id: ShardId) -> Self {
+        let inner = InnerPrefetchStagingArea {
+            slots: SizeTrackedHashMap {
+                map: Default::default(),
+                size_bytes: 0,
+                metrics: StagedMetrics::new(shard_id),
+            },
+        };
+        Self(Arc::new(Mutex::new(inner)))
```

Comment:
Should `slots.update_metrics()` be called upon struct creation?

Reply:
done